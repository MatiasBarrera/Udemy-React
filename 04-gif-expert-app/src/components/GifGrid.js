import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifts(category)
  return (
    <>
      <h3 className="animate__animated animate__bounceIn">{category}</h3>

      {loading && <p className="animate__animated flash">Loading</p>}

      <div className="card-grid">
        {images.map(img => <GifGridItem key={img.id} {...img} />)}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid
