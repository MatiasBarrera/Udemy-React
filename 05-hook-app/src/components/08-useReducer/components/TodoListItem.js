import React from 'react'
import PropTypes from 'prop-types'

const TodoListItem = ({ todo, index, handleToggle, handleDelete }) => {
  return (
    <li key={todo.id} className="list-group-item">
      <p
        className={`${todo.done && 'complete'}`}
        onClick={() => handleToggle(todo.id)}>
        {index + 1}. {todo.desc}
      </p>
      <button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
        Borrar
      </button>
    </li>
  )
}

TodoListItem.propTypes = {
  todo: PropTypes.object,
  index: PropTypes.number,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default TodoListItem
