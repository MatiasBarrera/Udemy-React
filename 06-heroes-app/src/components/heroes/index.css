.card-columns {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  width: 400px;
  height: 300px;
  border: 1px solid lightgray;
  margin: 1rem 1rem 1rem 0rem;
  display: flex;
}

.card-img {
  width: 170px;
  height: 100%;
}

.card-body {
  display: flex;
  flex-direction: column;
  justify-content: stretch;
  padding: 2em 2em;
}

.card-text {
  padding: 0.5em 0em;
}

.text-muted {
  color: lightgray;
}

.link {
  color: blue;
  text-align: center;
  padding: 0.2em 0px;
  width: 100px;
  font-weight: 500;
  border: 1px solid blue;
  border-radius: 10px;
}

.link:hover {
  background-color: blue;
  transition: ease-in-out 0.1s;
  color: #fff;
}

/* HeroScreen */

.hero-container {
  display: flex;
  justify-content: center;
  margin-top: 3rem;
}

.hero-image {
  border: 1px solid lightgray;
  padding: 1em 1em;
  width: 400px;
  height: 600px;
}

.hero-image img {
  width: 100%;
  height: 100%;
}

.hero-detail {
  border: 1px solid lightgray;
  padding: 1em 1em;
  width: 500px;
  margin-left: 1em;
}

.list-group {
  list-style: none;
  padding: 1em 1em;
}

.list-group li {
  margin-bottom: 10px;
}

.btn-return {
  margin-top: 3rem;
  border: 2px solid red;
  padding: 0.5em 2em;
  background-color: transparent;
  color: red;
  font-weight: bolder;
  border-radius: 10px;
  cursor: pointer;
}

.btn-return:hover {
  color: #fff;
  background-color: red;
  transition: ease-in-out 0.1s;
}

/* Search */

.search-container {
  margin-top: 3rem;
}

.search-form {
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.search-form h4 {
  padding-right: 1em;
}

.search-form form {
  display: flex;
  justify-content: space-between;
}

.search-form input {
  border: none;
  border-bottom: 2px solid lightgray;
  padding: 1em 1em;
  width: 29rem;
  font-size: medium;
}

.search-form button {
  width: 2.5em;
  font-size: 1.5em;
  border: none;
  background-color: transparent;
  cursor: pointer;
}

.search-form button:hover {
  border: 1px solid lightblue;
  border-radius: 10px;
  transition: ease-in-out 0.1s;
}

.search-result {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.alert {
  border: 1px solid lightgrey;
  width: 50%;
  text-align: center;
  padding: 1em 0em;
  border-radius: 10px;
  margin-top: 1rem;
  font-size: 1.2rem;
  color: #fff;
}

.alert-info {
  background-color: lightskyblue;
}

.alert-danger {
  background-color: tomato;
}
