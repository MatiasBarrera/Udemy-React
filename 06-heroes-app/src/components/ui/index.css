nav {
  background-color: #082032;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  padding: 20px;
  width: 100vw;
  height: 5%;
}

.brand {
  color: #ff4c29;
  display: flex;
  font-size: 25px;
  justify-content: center;
}

.container-links {
  grid-column: 3/3;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.container-logout {
  grid-column: 7/7;
  align-self: center;
  justify-self: center;
}

.container-logout button {
  background-color: transparent;
  border: 1px solid lightgray;
  color: white;
  cursor: pointer;
  font-weight: bolder;
  padding: 0.3rem 2rem;
}

.container-logout button:hover {
  background-color: white;
  color: black;
  transition: all 0.5s;
}

.container-user {
  grid-column: 6/7;
  align-self: center;
  justify-self: flex-end;
  color: #ff4c29;
  font-weight: bold;
}

a {
  text-decoration: none;
  color: #fff;
  font-weight: bold;
}
