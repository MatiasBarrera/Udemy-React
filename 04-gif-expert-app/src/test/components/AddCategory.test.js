import { shallow } from "enzyme";
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en componente AddCategory.js', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories}/>)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories}/>)
  })

  test('Debe mostrar el componente <AddCategory /> correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  });

  test('Debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input')
    const value = 'Hola mundo'
    input.simulate('change', {target: { value }}) 
    const inputValueChanged = wrapper.find('input')
    
    expect(inputValueChanged.prop('value')).toBe(value)

  });

  test('No debe de postear la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} })
    expect(setCategories).not.toHaveBeenCalled()
  });

  test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
    const input = wrapper.find('input')
    const value = 'Pokemon'

    input.simulate('change', {target: { value }}) 
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalled()
    expect(wrapper.find('input').prop('value')).toBe('')
  });
});
