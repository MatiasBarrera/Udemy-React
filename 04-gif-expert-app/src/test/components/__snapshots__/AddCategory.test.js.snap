// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AddCategory.js Debe mostrar el componente <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
